pulsar-metadata/src/main/java/org/apache/pulsar/metadata/coordination/impl/LeaderElectionImpl.java

```diff
@@ -207,11 +219,6 @@ private synchronized CompletableFuture<LeaderElectionState> tryToBecomeLeader()
                         // There was a conflict between 2 participants trying to become leaders at same time. Retry
                         // to fetch info on new leader.
 
-                        // We force the invalidation of the cache entry. Since we received a BadVersion error, we
-                        // already know that the entry is out of date. If we don't invalidate, we'd be retrying the
-                        // leader election many times until we finally receive the notification that invalidates the
-                        // cache.
-                        cache.invalidate(path);
```

Comment:
This is taken care of in `elect` method. It uses `store.get` and by-passes the cache. There's `cache.refresh(path);` to invalidate the stale value.

Reply:
The `elect()` method only calls `cache.refresh(path);` after a callback from this method (`handleExistingLeaderValue()`) is completed, and if we don't invalidate the cache, I think we'll recursively call `handleExistingLeaderValue()` as the comment indicates because `store.get(path)` will return a non empty result.

Reply:
It's sufficient to call `cache.refresh(path)` after the election has been completed.
The cached value is not used in any away in the leader election code path. That's why I don't think it would make any difference. `cache.get` is currently called in `tryToBecomeLeader`, but that seems unnecessary and should be removed.